packages/eslint-plugin/src/rules/keyword-spacing.ts

```diff
@@ -0,0 +1,598 @@
+          throw new Error(
+            'Failed to find token get, set, or async beside method name',
+          );
+        }
+
+        checkSpacingAround(token);
+      }
+    }
+
+    /**
+     * Reports `await` keyword of a given node if usage of spacing before
+     * this keyword is invalid.
+     * @param node A node to report.
+     */
+    function checkSpacingForAwaitExpression(node: TSESTree.AwaitExpression) {
+      checkSpacingBefore(sourceCode.getFirstToken(node)!);
+    }
+
+    /**
+     * Reports `as` keyword of a given node if usage of spacing before
+     * this keyword is invalid.
+     * @param node A node to report.
+     */
+    function checkSpacingForAsExpression(node: TSESTree.TSAsExpression) {
+      const token = sourceCode.getTokenAfter(node.expression)!; // get the `as` identifier.
+      checkSpacingAround(token);
+    }
+
+    return {
+      // Statements
+      DebuggerStatement: checkSpacingAroundFirstToken,
+      WithStatement: checkSpacingAroundFirstToken,
+
+      // Statements - Control flow
+      BreakStatement: checkSpacingAroundFirstToken,
+      ContinueStatement: checkSpacingAroundFirstToken,
+      ReturnStatement: checkSpacingAroundFirstToken,
+      ThrowStatement: checkSpacingAroundFirstToken,
+      TryStatement: checkSpacingForTryStatement,
+
+      // Statements - Choice
+      IfStatement: checkSpacingForIfStatement,
+      SwitchStatement: checkSpacingAroundFirstToken,
+      SwitchCase: checkSpacingAroundFirstToken,
+
+      // Statements - Loops
+      DoWhileStatement: checkSpacingForDoWhileStatement,
+      ForInStatement: checkSpacingForForInStatement,
+      ForOfStatement: checkSpacingForForOfStatement,
+      ForStatement: checkSpacingAroundFirstToken,
+      WhileStatement: checkSpacingAroundFirstToken,
+
+      // Statements - Declarations
+      ClassDeclaration: checkSpacingForClass,
+      ExportNamedDeclaration: checkSpacingForModuleDeclaration,
+      ExportDefaultDeclaration: checkSpacingForModuleDeclaration,
+      ExportAllDeclaration: checkSpacingForModuleDeclaration,
+      FunctionDeclaration: checkSpacingForFunction,
+      ImportDeclaration: checkSpacingForModuleDeclaration,
+      VariableDeclaration: checkSpacingAroundFirstToken,
+
+      // Expressions
+      ArrowFunctionExpression: checkSpacingForFunction,
+      AwaitExpression: checkSpacingForAwaitExpression,
+      ClassExpression: checkSpacingForClass,
+      FunctionExpression: checkSpacingForFunction,
+      NewExpression: checkSpacingBeforeFirstToken,
+      Super: checkSpacingBeforeFirstToken,
+      ThisExpression: checkSpacingBeforeFirstToken,
+      UnaryExpression: checkSpacingBeforeFirstToken,
+      YieldExpression: checkSpacingBeforeFirstToken,
+
+      // Others
+      ImportNamespaceSpecifier: checkSpacingForImportNamespaceSpecifier,
+      MethodDefinition: checkSpacingForProperty,
+      Property: checkSpacingForProperty,
+      TSAsExpression: checkSpacingForAsExpression,
```

Comment:
Real addition here.

Reply:
If this is the only addition (along with it's function), you hopefully should be able to just re-use the base rule, and merge your addition in with the rule.

Check out [`@typescript-eslint/quotes`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/quotes.ts) & [`@typescript-eslint/semi`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/semi.ts) for examples of how to do this.

(You still might have to duplicate some code in, but it should at least remove the need to duplicate this return).

Reply:
Nice! This simplifies a lot. However I still had to redefine most helper functions, since they are not exported by the eslint rule.

Reply:
Awesome! While I expected you'd have to duplicate some of the helpers, it's annoying that there isn't any node that calls the methods you want that we could trick into doing the work for us, but oh well it's still a nice simplification 🤷‍♂

Reply:
note that there is already the keyword `as` registered in the rule due to `import { x as y } from 'foo'`.
Looking at the rule's source code, you might be able to get away with just this fudge code for this extension:

```ts
return {
  ...baseRules,
  TSAsExpression(node): void {
    baseRules.ImportNamespaceSpecifier(node as any);
  },
};
```